example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx

```diff
@@ -0,0 +1,139 @@
+    $loadItem: Boolean!
+  ) {
+    scr_FormWizardCompositionO2OTestEntityById(id: $id)
+      @include(if: $loadItem) {
+      id
+      _instanceName
+      name
+    }
+  }
+`;
+
+const UPSERT_SCR_FORMWIZARDCOMPOSITIONO2OTESTENTITY = gql`
+  mutation Upsert_scr_FormWizardCompositionO2OTestEntity(
+    $formWizardCompositionO2OTestEntity: inp_scr_FormWizardCompositionO2OTestEntity!
+  ) {
+    upsert_scr_FormWizardCompositionO2OTestEntity(
+      formWizardCompositionO2OTestEntity: $formWizardCompositionO2OTestEntity
+    ) {
+      id
+    }
+  }
+`;
+
+const FormWizardCompositionO2O = observer(
+  (props: EntityEditorProps<FormWizardCompositionO2OTestEntity>) => {
+    const {
+      onCommit,
+      entityInstance,
+      submitBtnCaption = "common.submit"
+    } = props;
+
+    const [form] = useForm();
+
+    const {
+      executeLoadQuery,
+      loadQueryResult: { loading: queryLoading, error: queryError },
+      upsertMutationResult: { loading: upsertLoading },
+      serverValidationErrors,
+      intl,
+      handleSubmit,
+      handleSubmitFailed,
+      handleCancelBtnClick
+    } = useEntityEditor<FormWizardCompositionO2OTestEntity>({
+      loadQuery: LOAD_SCR_FORMWIZARDCOMPOSITIONO2OTESTENTITY,
+      upsertMutation: UPSERT_SCR_FORMWIZARDCOMPOSITIONO2OTESTENTITY,
+      entityName: ENTITY_NAME,
+      routingPath: ROUTING_PATH,
+      onCommit,
+      entityInstance,
+      useEntityEditorForm: createUseAntdForm(form),
+      useEntityEditorFormValidation: createUseAntdFormValidation(form)
+    });
+
+    if (queryLoading) {
+      return <Spinner />;
+    }
+
+    if (queryError != null) {
+      console.error(queryError);
+      return <RetryDialog onRetry={executeLoadQuery} />;
+    }
+
+    return (
+      <Card className="narrow-layout">
+        <Form
+          onFinish={handleSubmit}
+          onFinishFailed={handleSubmitFailed}
+          layout="vertical"
+          form={form}
+          validateMessages={createAntdFormValidationMessages(intl)}
+        >
+          <Field
+            entityName={ENTITY_NAME}
+            propertyName="name"
+            formItemProps={{
+              style: { marginBottom: "12px" }
+            }}
+          />
+
+          <GlobalErrorsAlert serverValidationErrors={serverValidationErrors} />
+
+          <Form.Item style={{ textAlign: "center" }}>
+            <Button htmlType="button" onClick={handleCancelBtnClick}>
+              <FormattedMessage id="common.cancel" />
+            </Button>
+            <Button
+              type="primary"
+              htmlType="submit"
+              loading={upsertLoading}
+              style={{ marginLeft: "8px" }}
+            >
+              <FormattedMessage id={submitBtnCaption} />
+            </Button>
+          </Form.Item>
+        </Form>
+      </Card>
+    );
+  }
+);
+
+registerEntityEditor({
+  component: FormWizardCompositionO2O,
+  caption: "screen.FormWizardCompositionO2O",
+  screenId: "FormWizardCompositionO2O",
```

Comment:
Clicking `Cancel` seems to cause a crash:

```
index.esm.js:8148 Uncaught TypeError: Cannot read property 'params' of null
    at Screens.getUrl (index.esm.js:8148)
    at Screens.<anonymous> (index.esm.js:8107)
    at executeAction (action.ts:65)
    at Screens.<unnamed action> [as setActiveScreen] (action.ts:46)
    at index.esm.js:572
    at handleClick (button.js:207)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994)
    at invokeGuardedCallback (react-dom.development.js:4056)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4070)
    at executeDispatch (react-dom.development.js:8243)
    at processDispatchQueueItemsInOrder (react-dom.development.js:8275)
    at processDispatchQueue (react-dom.development.js:8288)
    at dispatchEventsForPlugins (react-dom.development.js:8299)
    at react-dom.development.js:8508
    at batchedEventUpdates$1 (react-dom.development.js:22396)
    at batchedEventUpdates (react-dom.development.js:3745)
    at dispatchEventForPluginEventSystem (react-dom.development.js:8507)
    at attemptToDispatchEvent (react-dom.development.js:6005)
    at dispatchEvent (react-dom.development.js:5924)
    at unstable_runWithPriority (scheduler.development.js:468)
    at runWithPriority$1 (react-dom.development.js:11276)
    at discreteUpdates$1 (react-dom.development.js:22413)
    at discreteUpdates (react-dom.development.js:3756)
    at dispatchDiscreteEvent (react-dom.development.js:5889)
```

Reply:
Also only happens in `Form Wizard Editor`.

Reply:
Both issues are likely to be unrelated to entity wizard itself and related to screens API. Probably its best to create a separate issue for that.

Reply:
Yes, it doesn't work for other editors either. I created an issue:
https://github.com/Haulmont/jmix-frontend/issues/509